drivers/pcie/host/pcie.c

```diff
@@ -46,26 +46,51 @@ void pcie_set_cmd(pcie_bdf_t bdf, uint32_t bits, bool on)
 	pcie_conf_write(bdf, PCIE_CONF_CMDSTAT, cmdstat);
 }
 
-uintptr_t pcie_get_mbar(pcie_bdf_t bdf, unsigned int index)
+struct pci_mbar pcie_get_mbar(pcie_bdf_t bdf, unsigned int index)
 {
 	uint32_t reg, bar;
-	uintptr_t addr = PCIE_CONF_BAR_NONE;
-
-	reg = PCIE_CONF_BAR0;
-	for (bar = 0; bar < index && reg <= PCIE_CONF_BAR5; bar++) {
-		if (PCIE_CONF_BAR_64(pcie_conf_read(bdf, reg++))) {
-			reg++;
+	struct pci_mbar mbar = {0};
+
+	bar = 0;
+	for (reg = PCIE_CONF_BAR0; reg <= PCIE_CONF_BAR5; reg++) {
```

Comment:
you'll need to update the documentation of pcie_get_mbar() telling it skips useless bars, and index actually points to valid ones only.
btw, put the bar = 0 in the the for() initialization place, and bar++ next to reg++ 
It'll be easier to read.